ddprof-ffi/src/lib.rs

```diff
@@ -104,6 +104,12 @@ pub struct Slice<'a, T> {
     phantom: PhantomData<&'a [T]>,
 }
 
+// Use to represent strings -- should be valid UTF-8.
+type CharSlice<'a> = crate::Slice<'a, c_char>;
+
+/// Use to represent bytes -- does not need to be valid UTF-8.
+type ByteSlice<'a> = crate::Slice<'a, u8>;
```

Comment:
I'm not against it per se, but -- why should libddprof care if it is given non-utf-8 data? What are we gaining by distinguishing both types?

Reply:
Datadog cares about UTF-8 in certain cases, like tag names. Generally we need UTF-8 for string like thing or else it's probably garbage. However, certain things are not UTF-8, like many binary data formats. I think it's important that we separate things that are  bytes and not characters in the human sense of things.

Reply:
That's fair enough I guess. I think we could probably handle non-utf8-ness at other places downstream, but I think either way is fine, and we can always revisit as needed. 👍